fdbserver/VersionedBTree.actor.cpp

```diff
@@ -2227,39 +2265,60 @@ class DWALPager final : public IPager2 {
 		if (exists && fileSize >= (self->smallestPhysicalBlock * 2)) {
 			debug_printf("DWALPager(%s) recovering using existing file\n", self->filename.c_str());
 
-			state bool recoveredHeader = false;
-
-			// Read physical page 0 directly
-			wait(store(self->headerPage, self->readHeaderPage(self, 0)));
-
-			// If the checksum fails for the header page, try to recover committed header backup from page 1
-			if (!self->headerPage->verifyChecksum(0)) {
-				TraceEvent(SevWarn, "RedwoodRecoveringHeader").detail("Filename", self->filename);
+			state bool recoveredBackupHeader = false;
 
-				wait(store(self->headerPage, self->readHeaderPage(self, 1)));
+			// Try to read primary header
+			try {
+				wait(store(self->headerPage, self->readHeaderPage(primaryHeaderPageID)));
+			} catch (Error& e) {
+				debug_printf("DWALPager(%s) Primary header read failed with %s\n", self->filename.c_str(), e.what());
```

Comment:
just curious, any reason we need both TraceEvent(Warn) and debug_printf in this case?

Reply:
Most Redwood debugging is done with stdout. it's just far easier that way as there is an insane amount of detail and ordering is important so it's a pain to merge it with trace logs.  `debug_printf` compiles to nothing unless `REDWOOD_DEBUG` is set.